gp-includes/routes/glossary-entry.php

```diff
@@ -81,15 +81,25 @@ public function glossary_entry_add_post( $project_path, $locale_slug, $translati
 			$this->redirect( gp_url_join( gp_url_project_locale( $project->path, $locale_slug, $translation_set_slug ), array( 'glossary' ) ) );
 		}
 		else {
-			$created_glossary_entry = GP::$glossary_entry->create_and_select( $new_glossary_entry );
+			$find_parms = array(
+				'glossary_id' => $glossary->id,
+				'term' => $new_glossary_entry->term,
+			);
 
-			if ( ! $created_glossary_entry ) {
-				$this->errors[] = __( 'Error in creating glossary entry!', 'glotpress' );
-				$this->redirect( gp_url_join( gp_url_project_locale( $project->path, $locale_slug, $translation_set_slug ), array( 'glossary' ) ) );
-			}
-			else {
-				$this->notices[] = __( 'The glossary entry was created!', 'glotpress' );
+			if ( GP::$glossary_entry->find_one( $find_parms ) ) {
+				// Translators: %s is the glossary term that was attempted to be added.
+				$this->errors[] = sprintf( __( 'Error "%s" is a duplicate glossary entry!', 'glotpress' ), esc_html( $new_glossary_entry->term ) );
```

Comment:
Maybe make that message similar to https://github.com/GlotPress/GlotPress-WP/blob/cb79407aec395b4e3a5f6e0f9b3a79eb91914168/gp-includes/routes/translation.php#L290